priv/priv.go

```diff
@@ -36,7 +36,8 @@ var (
 
 // Mutex is the snappy mutual exclusion primitive.
 type Mutex struct {
-	lock *FileLock
+	filename string
+	flock    LockedFile
```

Comment:
Misnomer or not, it's still not clear why we need these two APIs which do exactly the same thing in slightly different ways.

What about having a single one that looks like:

type FileLock struct {
        filename string
        fd            int
}

func NewFileLock(filename string) *FileLock
func (l *FileLock) Lock() error
func (l *FileLock) TryLock() error
func (l *FileLock) Unlock() error

and then moving the root-checking logic to very early on in the execution of the binary, perhaps inside withMutexAndRetry for the time being.  There seems to be a lot of duplicated boilerplate around these functions too, but that's a separate cleanup. I'm happy to look at that one after you do this initial consolidation, if you want.